ygnmi/types.go

```diff
@@ -199,10 +205,11 @@ type NonLeafWildcardQuery[T ygot.ValidatedGoStruct] struct {
 func (lq *NonLeafWildcardQuery[T]) isWildcard() {}
 
 type nonLeafBaseQuery[T ygot.ValidatedGoStruct] struct {
-	dir     string
-	state   bool
-	ps      PathStruct
-	yschema *ytypes.Schema
+	dir            string
+	state          bool
+	ps             PathStruct
+	subPathStructs []PathStruct
```

Comment:
What is a `subPathStructs`? Should we make this clearer to readers who are unfamiliar to the library? I'm anticipating many readers of this code will be unfamiliar with most of the codebase.

Also I'm wondering since these are all the paths we're querying, should it reflect that in its name? subpath could indicate anything, while something like "queryPathStructs" might makes this aspect clearer.

Reply:
add comments